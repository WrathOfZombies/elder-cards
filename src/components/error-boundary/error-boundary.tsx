import React from "react";
import { ErrorBoundary as ReactErrorBoundary } from "react-error-boundary";

const onError = (error: Error, info: { componentStack: string }) => {
  // TODO: Hook into telemetry here
  console.groupCollapsed(error.message);
  console.info(info);
  console.groupEnd();
};

const ErrorFallback: React.FC = () => (
  <div id="error" role="alert" aria-label="Something went wrong">
    <div className="error-robot" />
  </div>
);

/**
 * Renders an error boundary. We use native elements as this is
 * above the Fluent provider tree
 */
export const ErrorBoundary: React.FC = ({ children }) => (
  <ReactErrorBoundary FallbackComponent={ErrorFallback} onError={onError}>
    {children}
  </ReactErrorBoundary>
);
