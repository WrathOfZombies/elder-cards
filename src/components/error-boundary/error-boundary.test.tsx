import { shallow } from "enzyme";
import React from "react";
import { ErrorBoundary } from "./error-boundary";

describe("Testing ErrorBoundary", () => {
  test("should render a fallback component", () => {
    const wrapper = shallow(<ErrorBoundary />);
    const FallbackComponent = wrapper.prop("FallbackComponent");

    expect(shallow(<FallbackComponent />).html()).toMatchSnapshot();
  });
});
