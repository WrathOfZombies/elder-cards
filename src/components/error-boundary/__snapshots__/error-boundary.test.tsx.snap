// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ErrorBoundary should render a fallback component 1`] = `"<div id=\\"error\\" role=\\"alert\\" aria-label=\\"Something went wrong\\"><div class=\\"error-robot\\"></div></div>"`;
